File: backends/fiber/internal/models/step_ingredient.go
package models

import "github.com/google/uuid"

type StepIngredient struct {
	ID            uuid.UUID   `gorm:"column:step_ingredient_id;primaryKey;default:gen_random_uuid()" json:"id"`
	StepID        uuid.UUID   `gorm:"column:step_id" json:"-"`
	IngredientID  uuid.UUID   `gorm:"column:ingredient_id" json:"-"`
	Amount        float64     `gorm:"column:amount;not null" json:"amount"`
	AmountMetric  float64     `gorm:"column:amount_metric;not null" json:"amount_metric"`
	MeasurementID uuid.UUID   `gorm:"column:measurement_id" json:"-"`
	Measurement   Measurement `json:"measurement"`
}

File: backends/aspnet/Recipes.API/entities/StepIngredient.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Recipes.API.entities;

public partial class StepIngredient
{
    [JsonPropertyName("id")]
    public Guid StepIngredientId { get; set; }

    public Guid? StepId { get; set; }

    public Guid? IngredientId { get; set; }

    public decimal Amount { get; set; }

    public decimal AmountMetric { get; set; }

    public Guid? MeasurementId { get; set; }

    public virtual Ingredient? Ingredient { get; set; }

    public virtual Measurement? Measurement { get; set; }

    public virtual Step? Step { get; set; }
}
